src/components/Search/SearchOverTime.tsx

```diff
@@ -0,0 +1,178 @@
+import InputLabel from '@mui/material/InputLabel';
+import Tooltip from '@mui/material/Tooltip';
+import { cssRule } from 'typestyle';
+
+import { compareView } from '../../common/constants';
+import { useAppSelector } from '../../hooks/app';
+import {
+  Spacing,
+  DropDownMenuRaw,
+  DropDownItemRaw,
+  //SearchStyles can be found in CompareCards.ts
+  SearchStyles,
+} from '../../styles';
+import { RevisionsList } from '../../types/state';
+import SearchDropdown from './SearchDropdown';
+import SearchInput from './SearchInput';
+import SearchResultsList from './SearchResultsList';
+
+interface SearchProps {
+  selectLabel: string;
+  searchResults: RevisionsList[];
+  tooltip: string;
+  inputPlaceholder: string;
+  isEditable: boolean;
+}
+
+function SearchComponent({
+  selectLabel,
+  tooltip,
+  inputPlaceholder,
+  isEditable,
+  searchResults,
+}: SearchProps) {
+  const mode = useAppSelector((state) => state.theme.mode);
+  const styles = SearchStyles(mode);
+  const [displayDropdown, setDisplayDropdown] = useState(false);
+  //temporary until next PR covers selected revisions
+  const handleSearchResultsEditToggle = (toggleArray: RevisionsList[]) => {
+    console.log('handleSearchResultsEditToggle', toggleArray);
+  };
+
+  const displayedRevisions = {
+    revs: [],
+    repos: [],
+  };
+
+  /* These overriding rules update the theme mode by accessing the otherwise inaccessible MUI tooltip styles */
+  cssRule('.MuiPopover-root', {
+    $nest: {
+      '.MuiPaper-root': {
+        flexDirection: 'column',
+        ...(mode === 'light' ? DropDownMenuRaw.Light : DropDownMenuRaw.Dark),
+        $nest: {
+          '.MuiList-root': {
+            padding: `${Spacing.Small}px ${Spacing.xSmall}px`,
+            $nest: {
+              '.MuiMenuItem-root': {
+                ...(mode === 'light'
+                  ? DropDownItemRaw.Light
+                  : DropDownItemRaw.Dark),
+              },
+            },
+          },
+        },
+      },
+    },
+  });
+
+  const handleDocumentMousedown = useCallback(
+    (e: MouseEvent) => {
+      if (!displayDropdown) {
+        return;
+      }
+      const target = e.target as HTMLElement;
+      if (target.closest('.new-search-input') === null) {
+        // Close the dropdown only if the click is outside the search input or one
+        // of it's descendants.
+        setDisplayDropdown(false);
+      }
+    },
+    [displayDropdown],
+  );
+
+  const handleEscKeypress = (e: KeyboardEvent) => {
+    if (e.key === 'Escape') {
+      setDisplayDropdown(false);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleDocumentMousedown);
+    return () => {
+      document.removeEventListener('mousedown', handleDocumentMousedown);
+    };
+  }, [handleDocumentMousedown]);
+
+  useEffect(() => {
+    document.addEventListener('keydown', handleEscKeypress);
+    return () => {
+      document.removeEventListener('keydown', handleEscKeypress);
+    };
+  });
```

Comment:
If we want to run this at mount time, we need to pass an empty dependency list to `useEffect`. Otherwise this will be run at every render.

Even more correct would be to use `handleEscKeypress` as dependency, but then you'll need to use `useCallback` for it too, with an empty dependency list.
Or define `handleEscKeypress` directly inside the effect (like suggested above for `handleDocumentMousedown`.